include/openthread/border_agent.h

```diff
@@ -208,9 +208,15 @@ otError otBorderAgentSetId(otInstance *aInstance, const otBorderAgentId *aId);
  * Setting the ephemeral key again before a previously set key has timed out will replace the previously set key and
  * reset the timeout.
  *
- * While the timeout interval is in effect, the ephemeral key can be used only once by an external commissioner to
- * connect. Once the commissioner disconnects, the ephemeral key is cleared, and the Border Agent reverts to using
- * PSKc.
+ * During the timeout interval, the ephemeral key can be used only once by an external commissioner to establish a
+ * connection. After the commissioner disconnects, the ephemeral key is cleared, and the Border Agent reverts to
+ * using PSKc. If the timeout expires while a commissioner is still connected, the session will be terminated, and the
+ * Border Agent will cease using the ephemeral key and revert to PSKc. The ephemeral key timeout cannot be extended by
+ * the commissioner (candidate) sending keep-alive messages (COMM_KA.req).
+ *
+ * Regardless of whether the secure connection is established using an ephemeral key or PSKc, the general timeout
+ * mechanism, using `TIMEOUT_COMM_PET` (50 seconds) to disconnect from stale sessions, still applies. This 50-second
+ * timeout can be extended by the commissioner (candidate) sending keep-alive messages.
```

Comment:
```suggestion
 * timeout can be extended by the commissioner sending keep-alive messages.
```

The candidate can't extend the 50s timeout if it does not petition as active commissioner.

Reply:
I believe the current implementation allows for this behavior. Anytime a Keep-alive is received, the timer restarts. The only check is that `mState` is not `kStateStopped`:
https://github.com/openthread/openthread/blob/d015004a74d8386f74894783c7e662c8d892e73e/src/core/meshcop/border_agent.cpp#L493-L503

Perhaps this needs clarification within the specification and an update, but that would be a topic for a separate PR.

I'll remove the `(candidate)` in the comment, so we can hopefully move forward with this PR. We've addressed and discussed several unrelated topics here (like adding more clarification comments).

Reply:
After second thought, I think it might still be valuable to directly callout the DTLS session can't be extended by the candidate once established, along with the EphemeralKey APIs as the suggestions.

As though the change itself doesn't change the logic regarding the 50s, along with introduction of epskc, candidate might be a more general scenario for credentials extraction.